ArduCopter/crash_check.cpp

```diff
@@ -49,13 +51,27 @@ void Copter::crash_check()
         return;
     }
 
+    // check for lean angle over 15 degrees
+    const float lean_angle_deg = degrees(acosf(ahrs.cos_roll()*ahrs.cos_pitch()));
+    if (lean_angle_deg <= CRASH_CHECK_ANGLE_MIN_DEG) {
```

Comment:
This is the fast loop.  We could avoid a chunk of maths here by doing the fancy maths operations on `CRASH_CHECK_ANGLE_MIN_DEG` rather than on the AHRS values.

Something like:
```suggestion
    static constexpr threshold = cosf(radians(CRASH_CHECK_ANGLE_MIN_DEG));
    if (ahrs.cos_roll()*ahrs.cos_pitch() <= threshold) {
```

... but compiling and actually working.. I'm not sure if a fabsf might be required there or not.

Reply:
I tested this suggestion (after adding "float" so that it compiles) and it produced the reverse of what I would expect.  I.e. it always said the angle was over the threshold when it wasn't and vice-versa.  I could take a poke at reversing it but this change was the one suggested by Leonard so I think I'll merge it instead of taking the chance.